<?xml version="1.0"?>

<robot name="sr_scara_hx" xmlns:xacro="http://www.ros.org/wiki/xacro">

<xacro:property name="mass" value="1" /> 	  <!-- arbitrary value for mass -->
<xacro:property name="pos0" value="0.00158925" /> <!-- Link 1 -->
<xacro:property name="pos2" value="0.300" />      <!-- Link 2 -->
<xacro:property name="pos3" value="0" />          <!-- Link 3 -->


  <!-- Import all Gazebo-customization elements, including Gazebo colors -->
  <xacro:include filename="$(find robot_simulation)/URDF/sr_scara_hx.gazebo" />
  <!-- Import Rviz colors -->
  <xacro:include filename="$(find robot_simulation)/URDF/materials.xacro" />

<!--Links -->
<link name="base_link">
 <visual>
  <origin xyz="0 0 ${pos0}" rpy="0 0 0" />
   <geometry>
     <mesh filename="package://robot_simulation/meshes/base_R01.stl" scale="0.001 0.001 0.001"/>  
   </geometry>
   <material name="black"/>
 </visual>
 <collision>
  <origin xyz="0 0 ${pos0}" rpy="0 0 0" />
   <geometry>
     <mesh filename="package://robot_simulation/meshes/base_R01.stl" scale="0.001 0.001 0.001"/>  
   </geometry>
 </collision>  
 <inertial>
      <origin xyz="0 0 ${pos0}" rpy="0 0 0"/>
      <mass value="${mass}"/>
      <inertia
	  ixx="0.0" ixy="0.0" ixz="0.0"
	  iyy="0.0" iyz="0.0"
	  izz="0.0" />
 </inertial>
</link>

<link name="link1">
 <visual>
  <origin xyz="0 0 0" rpy="0 0 0" />
   <geometry>
     <mesh filename="package://robot_simulation/meshes/arm_link_R01.stl" scale="0.001 0.001 0.001"/>  
   </geometry>
   <material name="red"/>
 </visual>
 <collision>
  <origin xyz="0 0 0" rpy="0 0 0" />
   <geometry>
     <mesh filename="package://robot_simulation/meshes/arm_link_R01.stl" scale="0.001 0.001 0.001"/>  
   </geometry>
 </collision>
 <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="${mass}"/>
      <inertia
	  ixx="" ixy="0.0" ixz="0.0"
	  iyy="" iyz="0.0"
	  izz=""/>
 </inertial>
</link>

<link name="link2">
 <visual>
  <origin xyz="${pos2} 0 0" rpy="0 0 0" />
   <geometry>
     <mesh filename="package://robot_simulation/meshes/head_link_R01.stl" scale="0.001 0.001 0.001"/>  
   </geometry>
   <material name="black"/>
 </visual>
 <collision>
  <origin xyz="${pos2} 0 0" rpy="0 0 0" />
   <geometry>
     <mesh filename="package://robot_simulation/meshes/head_link_R01.stl" scale="0.001 0.001 0.001"/>  
   </geometry>
  </collision>
 <inertial>
      <origin xyz="${pos2} 0 0" rpy="0 0 0"/>
      <mass value="${mass}"/>
      <inertia
	  ixx="0.0" ixy="0.0" ixz="0.0"
	  iyy="0.0" iyz="0.0"
	  izz="0.0"/>
 </inertial>
</link>

<link name="probe">
 <visual>
  <origin xyz="${pos3} 0 0" rpy="0 0 0" />
  <geometry>
     <mesh filename="package://robot_simulation/meshes/End_link_R01.stl" scale="0.001 0.001 0.001"/>  
  </geometry>
  <material name="orange"/>
 </visual>
 <collision>
  <origin xyz="${pos3} 0 0" rpy="0 0 0" />
  <geometry>
     <mesh filename="package://robot_simulation/meshes/End_link_R01.stl" scale="0.001 0.001 0.001"/>  
  </geometry>
 </collision>
 <inertial>
      <origin xyz="${pos3} 0 0" rpy="0 0 0"/>
      <mass value="${mass}"/>
      <inertia
	  ixx="0.0" ixy="0.0" ixz="0.0"
	  iyy="0.0" iyz="0.0"
	  izz="0.0"/>
 </inertial>
</link>


<!-- Joints 
<link name = "gripper_base">
 <visual>
  <origin xyz ="0 0 -0.085" rpy="0 0 0" />
   <geometry>
     <box size="0.06 .02 .01"/>
   </geometry>
 </visual>
</link>
<link name = "gripper_dummy">
 <visual>
  <origin xyz ="0 0 -0.095" rpy="0 0 0" />
   <geometry>
     <box size="0.06 .02 .01"/>
   </geometry>
 </visual>
 </link>>

-->


<!-- Joints -->

<joint name="joint1" type="revolute">
  <axis xyz="0 0 1" /> 
  <limit effort="3.0" velocity="0.01" lower="-2.617" upper="2.617"/>
  <origin xyz="0 0 0.00158925" rpy="0 0 0" />
  <parent link="base_link" />
  <child link="link1" />
</joint>

<joint name="joint2" type="revolute">
 <axis xyz="0 0 1" /> 
 <limit effort="3.0" velocity="0.01" lower="-2.617" upper="2.617"/>
 <origin xyz="-0.3 0 0" rpy="0 0 0" />
 <parent link="link1" />
 <child link="link2" />
</joint>

<joint name="joint3" type="prismatic">
 <axis xyz="0 0 1" /> 
 <limit effort="3.0" velocity="0.01" lower="-0.1" upper="0.05"/>
 <origin xyz="0.3 0 0" rpy="0 0 0" />
 <parent link="link2" />
 <child link="probe" />
</joint>


<!-- Joints 
<joint name ="gripper_con" type="revolute">
 <axis xyz="0 0 1" /> 
 <limit effort="3.0" velocity="0.01" lower="-3" upper="3"/>
 <origin rpy="0 0 0" xyz="0 0 0"/>
 <parent link = "probe"/>
 <child link = "gripper_base" />
</joint>

<joint name ="gripper_dummy" type="fixed">
 <origin rpy="0 0 0" xyz="0 0 0"/>
 <parent link = "gripper_base"/>
 <child link = "gripper_dummy" />
</joint>
-->

<!-- Transmissions -->

  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint1"/>
    <actuator name="motor1">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint2"/>
    <actuator name="motor2">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran3">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint3"/>
    <actuator name="motor3">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

</robot> 
